src/approx_const.rs

```diff
@@ -52,22 +52,41 @@ impl LateLintPass for ApproxConstant {
 
 fn check_lit(cx: &LateContext, lit: &Lit, e: &Expr) {
     match lit.node {
-        LitFloat(ref str, TyF32) => check_known_consts(cx, e, str, "f32"),
-        LitFloat(ref str, TyF64) => check_known_consts(cx, e, str, "f64"),
-        LitFloatUnsuffixed(ref str) =>
-            check_known_consts(cx, e, str, "f{32, 64}"),
+        LitFloat(ref s, TyF32) => check_known_consts(cx, e, s, "f32"),
+        LitFloat(ref s, TyF64) => check_known_consts(cx, e, s, "f64"),
+        LitFloatUnsuffixed(ref s) =>
+            check_known_consts(cx, e, s, "f{32, 64}"),
         _ => ()
     }
 }
 
 fn check_known_consts(cx: &LateContext, e: &Expr, s: &str, module: &str) {
-    if let Ok(value) = s.parse::<f64>() {
-        for &(name, lower_bound, upper_bound) in KNOWN_CONSTS {
-            if (value >= lower_bound) && (value < upper_bound) {
+    if let Ok(_) = s.parse::<f64>() {
+        for &(constant, name, min_digits) in KNOWN_CONSTS {
+            if is_approx_const(constant, s, min_digits) {
                 span_lint(cx, APPROX_CONSTANT, e.span, &format!(
                     "approximate value of `{}::{}` found. \
                     Consider using it directly", module, &name));
+                return;
             }
         }
     }
 }
+
+/// Returns false if the number of significant figures in `value` are
+/// less than `min_digits`; otherwise, returns true if `value` is equal
+/// to `constant`, rounded to the number of digits present in `value`.
+fn is_approx_const(constant: f64, value: &str, min_digits: usize) -> bool {
+    if value.len() <= min_digits {
+        false
+    } else {
+        let round_const = format!("{:.*}", value.len() - 2, constant);
+
+        let mut trunc_const = constant.to_string();
```

Comment:
I think using a slice instead of a copy would suffice, like `constant[0..value.len()]`.